### DISCUSSION TITLE
Error with CLONE option

### DISCUSSION THREAD

**Discussion by bjmarty122** [USNavalResearchLaboratory/mgen#53] [General]
I'm attempting the following mgen event "0.0 ON 1 UDP DST <IP address/Port> CLONE [tcpdump testfile.dat [0]]" and I receive the error "Error: invalid event option: CLONE"

I have version 5.02b of MGEN which should include the clone pattern option based on the documentation. Does anyone know why I am getting this error and how to fix it? Thank you.

**Comment by weston-nrl** [comment]
Where did you get 5.02b from?  It's a rather old version at this point, released in 2015.

If you are getting this as a binary from some repository (e.g. Ubuntu apt-get), then that version was most likely built without libpcap support, which is required for CLONE.  I would suggest downloading the latest source from here and building it yourself.  You will need the libpcap development libraries installed for your system (you can install those via apt-get, etc.).

**Reply by bjmarty122** [reply]
You are correct, I installed mgen using "sudo apt-get install mgen".

I attempted the following to install 5.1.1 after purging the previous install:
sudo apt-get install libpcap-dev

git clone https://github.com/USNavalResearchLaboratory/mgen.git
cd mgen
git submodule update --init
cd makefiles
make -f Makefile.linux mgen

This was unsuccessful and I received the error "bash: usr/bin/mgen  No such file or directory"

For reference, I am attempting to run this on a Pi using the latest Raspberry Pi OS

**Comment by bebopagogo** [comment]
Our MGEN makefiles build the binary executable but don’t install it anywhere.  So after building, you would need to copy it some location in your PATH that your bash shell searches for executable.  Since you are building it as a “local” add-on package, you may wish to install it to /usr/local/bin so, after building with the make command, you would need to issue

 

sudo cp mgen /usr/local/bin/

 

This assumes your account is in the sudo “wheel” and you can get the admin privileges to install into /usr/local/bin.  If you haven’t used “sudo”, just so you know it will prompt you for your password to validate your administrative access to perform that action.

 

At some point, we should update our build files to include an install option but haven’t done that yet.

 
Brian
 

From: bjmarty122 ***@***.***>
Reply-To: USNavalResearchLaboratory/mgen ***@***.***>
Date: Wednesday, January 24, 2024 at 3:52 PM
To: USNavalResearchLaboratory/mgen ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [USNavalResearchLaboratory/mgen] Error with CLONE option (Discussion #53)

 

You are correct, I installed mgen using "sudo apt-get install mgen".

I attempted the following to install 5.1.1 after purging the previous install:
sudo apt-get install libpcap-dev

git clone https://github.com/USNavalResearchLaboratory/mgen.git
cd mgen
git submodule update --init
cd makefiles
make -f Makefile.linux mgen

This was unsuccessful and I received the error "bash: usr/bin/mgen No such file or directory"

For reference, I am attempting to run this on a Pi using the latest Raspberry Pi OS

**Reply by bjmarty122** [reply]
Brian,

That was exactly what I needed, thank you! I know have 5.1.1 running and clone is working.
